### PR TITLE
feat: compassWallet connector

### PR DESCRIPTION
**Pull request by hsnice16**
Sei EVM is in continuous development, and so is Compass. 
That's why it has a link for a Canary version today.

### CODE REVIEW

**Comment by hsnice16** [comment]
It will work. Actually, It's live on DragonSwap - https://test.dragonswap.app/swap
Screenshot 2024-03-08 at 10 00 06 PM

DragonSwap team had faced an issue while integrating this, that's why I have raised a PR so other dApps (we are reaching too), can add it easily.
Screenshot 2024-03-08 at 10 00 58 PM

It was not working with `flag`, that's why I asked them to change that to `getInjectedConnector({ target: window.compassEvm })`

File: packages/rainbowkit/src/index.ts
Comment (DanielSinclair) [inline packages/rainbowkit/src/index.ts:7]: Shouldn't need to export this

File: packages/rainbowkit/src/wallets/walletConnectors/compassWallet/compassWallet.ts
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/compassWallet/compassWallet.ts:12]: I think we need to wait until this wallet is released in production

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

File: packages/rainbowkit/src/wallets/walletConnectors/compassWallet/compassWallet.ts
Comment (hsnice16) [reply packages/rainbowkit/src/wallets/walletConnectors/compassWallet/compassWallet.ts:12]: This wallet is related to Sei EVM development, and as that is something in continuous development, this wallet is also in development. That's why a canary version, so dApp can easily integrate and test out the transactions.

May I know (if it's okay), why do we need to wait till production?

Comment (hsnice16) [reply packages/rainbowkit/src/wallets/walletConnectors/compassWallet/compassWallet.ts:12]: I was also thinking, that I will need to raise one more PR once Sei EVM goes live, as then this wallet will be in production.

File: packages/rainbowkit/src/index.ts
Comment (hsnice16) [reply packages/rainbowkit/src/index.ts:7]: There was a discussion on this - https://github.com/rainbow-me/rainbowkit/discussions/1835#discussion-6338498

Comment (hsnice16) [reply packages/rainbowkit/src/index.ts:7]: If you check the screenshot shared in the above message - https://github.com/rainbow-me/rainbowkit/pull/1837#issuecomment-1986014696, the dApp was not able to create a wallet connect as this function was not exposed to the user.

**Comment by hsnice16** [comment]
@magiziz @DanielSinclair please do review this.

**Comment by magiziz** [comment]
@hsnice16 Could we wait until it's released to production ? The extension is currently marked as "canary" and only 5 users are using it only.

**Comment by hsnice16** [comment]
Yeah, as I have mentioned above in all the replies, this is still in development, and this PR will help dApps integrate it easily and do development faster, that's why it has fewer users and we will need you to merge this PR.

**Comment by magiziz** [comment]
> Yeah, as I have mentioned above in all the replies, this is still in development, and this PR will help dApps integrate it easily and do development faster, that's why it has fewer users and we will need you to merge this PR.

We usually prefer wallets working in production before merging, otherwise we'll have to maintain it with adding new links, new steps and just watch out for it. We'll have this PR open until they've released in production.

**Comment by hsnice16** [comment]
You can merge this PR for now, and once it goes live in production, I will raise one more PR, and after that, you won't have to do anything.

**Comment by DanielSinclair** [comment]
@hsnice16 We will wait for the time being so that we can properly QA. You can test out your wallet with this example deployment [here](https://rainbowkit-example-git-fork-hsnice16-feat-c-af646f-rainbowdotme.vercel.app/). Feel free to ping us here to re-open or open another PR when it gets further along, and then we'll review and merge.

**Comment by hsnice16** [comment]
Hey @magiziz @DanielSinclair, Sei EVM changes are live in production now. You can install compass wallet, and use the features - https://chromewebstore.google.com/detail/compass-wallet-for-sei/anokgmphncpekkhclmingpimjmcooifb

Let me know, what should be my next steps to add support for CompassWallet in Rainbowkit. 
Many teams developing in this space are asking for it, so it would be great if we could add it ASAP.

Thanks.

**Comment by magiziz** [comment]
@hsnice16 Sounds good. Will re-open this PR and re-review. Do you know if they support EVM chains by any chance ? I'm playing around with it rn, but doesn't seem like there is EVM support for the wallet.

**Comment by hsnice16** [comment]
Yes, they do. Sei EVM is live on Arctic-1 chain only, so you can switch to that chain. You can use DragonSwap dApp to test - https://test.dragonswap.app/swap

Let me know if you need anything.
https://github.com/rainbow-me/rainbowkit/assets/56081584/b391b6ac-5f51-4aed-96d3-a043f116d5a8

**Comment by hsnice16** [comment]
Let me know if you want me to raise a new PR (I am a Compass team member) @magiziz @DanielSinclair

**Comment by hsnice16** [comment]
team! @magiziz @DanielSinclair do let me know the next steps here.

**Comment by magiziz** [comment]
@hsnice16 Yes we'll come back to this as soon as we can. Please bear with us 🙏

**Comment by hsnice16** [comment]
Okay. Thanks, 👍

**Comment by hsnice16** [comment]
I have updated the Chrome link to the production Chrome web store link.
Let me know, if need any other change.

**Comment by magiziz** [comment]
@hsnice16 Sounds good. I'll take a last look and approve the PR.

**Comment by magiziz** [comment]
@hsnice16 Just did some changes for `compassWallet`, added wallet to the docs, included it in our example dApp + added sei chain and fixed some merge conflicts. Before mergin this can you make sure the `compassWallet.svg` is sort of rounded square instead of a transparent background ? I would prefer having the dark background and white compass logo like they have in their extension.

Example of dark mode image not working as expected:

example

**Comment by hsnice16** [comment]
Hey @magiziz, I updated the logo. You can check.

**Comment by hsnice16** [comment]
> chore: optimize compassWallet.svg

Hey @magiziz, what do you do in optimization?

**Comment by magiziz** [comment]
@hsnice16 I just optimized it with `svgo` library. I approved the PR we should be able to merge this either today or tomorrow.

**Review by 0xmkh — APPROVED** [review]

**Comment by hsnice16** [comment]
Thanks.

**Comment by hsnice16** [comment]
Do let me know if anything blocks the merging.

**Comment by magiziz** [comment]
@hsnice16 We'll merge this soon. Please bear with us 🙏

**Comment by hsnice16** [comment]
Okay.

**Review by DanielSinclair — APPROVED** [review]

**Comment by hsnice16** [comment]
thanks for merging this PR @DanielSinclair @magiziz 🙌
do we have any ETA on the next new package release?

**Comment by magiziz** [comment]
@hsnice16 Should be around today or tomorrow

**Comment by hsnice16** [comment]
Thanks

**Pull request merged**
